toolhub_auth/models.py

```diff
@@ -52,3 +59,28 @@ def __str__(self):
         if self.get_full_name():
             return self.get_full_name()
         return self.email
+
+    def get_absolute_url(self):
+        return reverse("profile", kwargs={"pk": self.pk})
+
+
+class UserProfile(TimeStampedModel):
+    """
+    Custom user profile page details
+    """
+
+    user = models.OneToOneField(User, on_delete=models.CASCADE, related_name="profile")
+    text = MarkdownxField(_("Profile Text"), blank=True)
+
+    def __str__(self):
+        return f"{self.user} profile"
+
+    def get_absolute_url(self):
+        return reverse("profile", kwargs={"pk": self.user_id})
+
+
+@receiver(post_save, sender=User)
```

Comment:
We've got a custom user model. Let's do a custom save method to keep the code with that model.